### PR TITLE
Add cassandra beaker backend using new cassandra driver

### PR DESCRIPTION
**Pull request by dbenamydd**
Hey Seth,

Once this is reviewed and we're happy with it, I'm going to close this PR unmerged and submit a new one to upstream. If they don't take it relatively soon, I'll time out and merge it to the datadog branch. If they do take it, is our workflow to update master to track upstream and then rebase datadog on top of it?

The tests are split apart this way so it's easy to run the same tests on multiple backends. I originally had the code to run them against the old cassandra_pycassa backend but it fails a bunch of them because of key listing nonsense and it didn't seem worth fixing.

Each commit stands alone.

### CODE REVIEW

File: beaker_extensions/_compat.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Mostly copy pasted from beaker.
```
Comment (dbenamydd) [inline beaker_extensions/_compat.py]: We don't use most of this. If you prefer I can only keep the parts that we do.

**Comment by sethrosenblum** [comment]
nitpick: can we call these cassandra_thrift and cassandra_cql ?

File: beaker_extensions/_compat.py (hunk 1 above)
Comment (sethrosenblum) [inline beaker_extensions/_compat.py]: That probably makes sense, why do we need any of it if it's in beaker?

Comment (dbenamydd) [inline beaker_extensions/_compat.py]: Will do.

It seems not be be installed, possibly because it's intended to be a ~private module.

**Comment by dbenamydd** [comment]
> nitpick: can we call these cassandra_thrift and cassandra_cql ?

Works for me.

Thanks for the great review!

**Comment by dbenamydd** [comment]
Ok, upstream didn't take it. I'll merge this.

**Comment by dbenamydd** [comment]
Oh actually, I think I just want to merge this to `datadog`.

**Pull request closed without merging**